pages/api/applications/[applicationId]/stages/index.ts

```diff
@@ -1,5 +1,133 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { Application, ApplicationStage, ApplicationStageType, PrismaClient } from '@prisma/client';
+import { ApiResponse, StatusMessageType } from '../../../../../types/apiResponse';
+import { createJsonResponse, HttpMethod, HttpStatus, rejectHttpMethod } from '../../../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../../../utils/prisma/prismaHelpers';
+import { ApplicationStageData, ApplicationStagePostData } from '../../../../../types/applicationStage';
+import { isValidDate } from '../../../../../utils/date/validations';
+import { withAuthUser } from '../../../../../utils/auth/jwtHelpers';
+import { Nullable } from '../../../../../types/utils';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const { ERROR, SUCCESS } = StatusMessageType;
+const prisma = new PrismaClient();
+
+type ApplicationDataWithStages = Application & { applicationStages: ApplicationStage[] };
+
+enum MessageType {
+  INVALID_TYPE,
+  INVALID_DATE,
+  INVALID_APPLICATION_ID,
+  APPLICATION_NOT_FOUND,
+  CREATED,
+}
+
+const messages = Object.freeze({
+  [MessageType.INVALID_TYPE]: {
+    type: ERROR,
+    message:
+      'Application Stage type is invalid. Type must be one of: ' + Object.values(ApplicationStageType).join(', ') + '.',
+  },
+  [MessageType.INVALID_DATE]: {
+    type: ERROR,
+    message: 'Application Stage date is invalid.',
+  },
+  [MessageType.INVALID_APPLICATION_ID]: {
+    type: ERROR,
+    message: 'Query parameter applicationId (.../applications/:applicationId/...) is invalid.',
+  },
+  [MessageType.APPLICATION_NOT_FOUND]: {
+    type: ERROR,
+    message: 'User does not have any application with the given ID.',
```

Comment:
Let's standardise to messages that are used for this message type in other files.

Reply:
Got it. I'm also going to change `/pages/api/applications/[applicationId].ts` `APPLICATION_DOES_NOT_BELONG_TO_USER` to match `APPLICATION_NOT_FOUND` because we should not reveal to a user whether the application exists for someone else.